### DISCUSSION TITLE
How to add consumer tags for `EventingBasicConsumer`?

### DISCUSSION THREAD

**Discussion by 64J0** [rabbitmq/rabbitmq-dotnet-client#1514] [Q&A] [answered]
Hello, hope you're good. I'd like to know what is the recommended way to add consumer tags to the [EventingBasicConsumer](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/RabbitMQ.Client/client/events/EventingBasicConsumer.cs#L44). I noticed that this constructor accepts only the `channel`, so I guess I'm using the wrong class.

What class should I use to have a similar behavior as the `EventingBasicConsumer` but with the possibility to set consumer tags?

**Comment by lukebakken** [comment]
Hello, thanks for using this library. Which version are you using?

**Reply by 64J0** [reply]
I'm using version `6.8.1`.

**Reply by 64J0** [reply]
I didn't mention before, but I'm using it with F# instead of C#.

**Comment by lukebakken** [comment, marked as answer]
Hello, sorry I'm just now getting back to this. Could you explain _why_ you need to set a specific consumer tag?

UPDATE - after looking at the code, you can specify your own consumer tag when you call `BasicConsume` on your `IModel` instance.

**Reply by michaelklishin** [reply]
We should clarify that there is no such thing as "adding a consumer tag to a consumer". It is possible to have multiple competing consumers on a queue that all use the same `Consumer` implementation instance (an object), which is what happens when `BasicConsumer` is used multiple times with the same object.

**Reply by 64J0** [reply]
Hey, just to clarify, I was trying to translate a different language implementation to .NET, and there they add this tag. Since I didn't find an example in .NET, I decided to open this discussion here.
